src/main/java/ai/chalk/client/ChalkClientImpl.java

```diff
@@ -74,8 +74,18 @@ public OnlineQueryResult onlineQuery(OnlineQueryParamsComplete params) throws Ch
                 .queryName(params.getQueryName())
                 .build();
 
-        // ignore the warning here, because we don't want to free the memory yet
-        return this.handler.sendRequest(request).toResult();
+        HttpResponse<byte[]> response = this.handler.sendRequest(request);
+        try (
+            var allocator = this.allocator.newChildAllocator(
```

Comment:
hmm, one complexity here since we're returning data to the client is that we can't actually close the allocator, since there are outstanding tables that they need to read...we probably should either:
1.  give them back the reference so they can close it themselves
2. give them back non-arrow data

Reply:
actually we could reassociate the returned data with the root allocator before we return it, so we get the limit during request handling and then can chose the child allocator

Reply:
1) is the best choice, discussed irl